examples/LLM/llama2/chat_app/torchserve_server_app.py

```diff
@@ -0,0 +1,200 @@
+        url = "http://localhost:8081/models/" + MODEL_NAME
+        res = requests.get(url)
+        if res.status_code != 200:
+            model_state_container.error("Error getting model status", icon="🚫")
+            return
+        status = json.loads(res.text)[0]
+        model_state_container.write(status)
+
+
+def scale_workers(workers):
+    if st.session_state.registered:
+        num_workers = st.session_state[workers]
+        url = (
+            "http://localhost:8081/models/"
+            + MODEL_NAME
+            + "?min_worker="
+            + str(num_workers)
+            + "&synchronous=true"
+        )
+        res = requests.put(url)
+        # if res.status_code != 200:
+        #    state_container.write(res.text)
+
+        server_state_container.caption(res.text)
+
+
+def set_batch_size(batch_size):
+    if st.session_state.registered:
+        url = "http://localhost:8081/models/" + MODEL_NAME + "/1.0"
+        res = requests.delete(url)
+        server_state_container.caption(res.text)
+        st.session_state.registered = False
+
+        batch_size = st.session_state[batch_size]
+        url = (
+            "http://localhost:8081/models?model_name="
+            + MODEL_NAME
+            + "&url="
+            + MODEL_NAME
+            + ".tar.gz"
+            + "&batch_size="
+            + str(batch_size)
+            + "&initial_workers="
+            + str(workers)
+            + "&synchronous=true"
+            + "&max_batch_delay="
+            + str(max_batch_delay)
+        )
+        _register_model(url)
+
+
+def set_max_batch_delay(max_batch_delay):
+    if st.session_state.registered:
+        url = "http://localhost:8081/models/" + MODEL_NAME + "/1.0"
+        res = requests.delete(url)
+        server_state_container.caption(res.text)
+        st.session_state.registered = False
+
+        max_batch_delay = st.session_state[max_batch_delay]
+        url = (
+            "http://localhost:8081/models?model_name="
+            + MODEL_NAME
+            + "&url="
+            + MODEL_NAME
+            + ".tar.gz"
+            + "&batch_size="
+            + str(batch_size)
+            + "&initial_workers="
+            + str(workers)
+            + "&synchronous=true"
+            + "&max_batch_delay="
+            + str(max_batch_delay)
+        )
+        _register_model(url)
+
+
+if "started" not in st.session_state:
+    st.session_state.started = False
+if "stopped" not in st.session_state:
+    st.session_state.stopped = False
+if "registered" not in st.session_state:
+    st.session_state.registered = False
+
+# Replicate Credentials
+with st.sidebar:
+    st.title("🦙💬 Llama 2 TorchServe Server ")
+
+    st.button("Start Server", on_click=start_server)
+    st.button("Stop Server", on_click=stop_server)
+    st.button("Register Llama2", on_click=register_model)
+    workers = st.sidebar.slider(
+        "Num Workers",
+        key="Num Workers",
+        min_value=1,
+        max_value=4,
```

Comment:
why max it out like this? I think you can make it up to the number of cores. If it crashes mgiht be cool to show how going to fewer workers also gives you error recovery

Reply:
Good point . Let me try and update the code/readme with description of what users can expect

Reply:
Seems to be running fine. My laptop is slow but nothing has crashed
<img width="1662" alt="Screenshot 2023-09-29 at 2 00 57 PM" src="https://github.com/pytorch/serve/assets/16617092/a5e2e220-b3eb-499b-9c0b-44a70acac961">